index.js

```diff
@@ -1,13 +1,14 @@
 var strftime = require('prettydate').strftime;
 
-module.exports = function generateCommonLog(request, response)
+module.exports = function generateCommonLog(request, response, options)
 {
     var protocol = 'HTTP/' + request.httpVersion;
     var payload_len = response._data ? response._data.length : '-';
     var UA = request.headers['user-agent'] || '-';
     var referer = request.headers['referer'] || '-';
     var tstamp = response._time ? new Date(response._time) : new Date();
-    var remote = (request.socket ? request.socket.remoteAddress : '') || '';
+    var remote = (options && options.ipHeader ? request.headers[options.ipHeader] : '') ||
```

Comment:
I like that you made this configurable :+1:, I think you might want to add to the check:

`(request.headers || {})[options.ipHeader]`

I recall seeing scenarios where `request.headers` is not set.